bijection-util/src/main/scala/com/twitter/bijection/twitter_util/UtilBijections.scala

```diff
@@ -122,6 +123,12 @@ trait UtilBijections {
       override def apply(pool: FuturePool) = new TwitterExecutionContext(pool)
       override def invert(context: ExecutionContext) = new ScalaFuturePool(context)
     }
+
+  implicit def byteArrayBufBijection: Bijection[Array[Byte], Buf] =
+    new AbstractBijection[Array[Byte], Buf] {
+      override def apply(bytes: Array[Byte]) = Buf.ByteArray.Owned(bytes)
```

Comment:
The [ChannelBuffer bijection](https://github.com/twitter/bijection/blob/71ed7b50d3ef4b5083060457dec6e52b6f65bad8/bijection-netty/src/main/scala/com/twitter/bijection/netty/ChannelBufferBijection.scala#L34) wraps the byte array on application and copies on extraction so this jives with the wrapping half but not the extraction. 

I can see a case for defining both versions and letting folks import the semantics they prefer? otoh, if users care about defensive copies, they can fairly trivially compose that onto any use of this version.

I do think making the shared impl the default is a mistake since this code tends to be used for serialization in batch jobs and as the hyper-doop folks showed is frequently a performance bottleneck.